assets/src/js/frontend/paypal-commerce/SmartButtons.js

```diff
@@ -21,36 +57,44 @@ class SmartButtons extends PaymentMethod {
 	 * @return {object} Smart button container selector.
 	 */
 	getButtonContainer() {
-		const smartButtonWrap = document.createElement( 'div' );
 		this.ccFieldsContainer = this.form.querySelector( '[id^="give_cc_fields-"]' ); // Refresh cc field container selector.
+		const oldSmartButtonWrap = this.ccFieldsContainer.querySelector( '#give-paypal-commerce-smart-buttons-wrap' );
+
+		if ( oldSmartButtonWrap ) {
+			return oldSmartButtonWrap;
+		}
 
-		smartButtonWrap.setAttribute( 'id', '#give-paypal-commerce-smart-buttons-wrap' );
+		const smartButtonWrap = document.createElement( 'div' );
+		const separator = this.ccFieldsContainer.querySelector( '.separator-with-text' );
+		smartButtonWrap.setAttribute( 'id', 'give-paypal-commerce-smart-buttons-wrap' );
+		const cardNumberWarp = this.ccFieldsContainer.querySelector( '[id^=give-card-number-wrap-]' );
 
-		return this.ccFieldsContainer.insertBefore( smartButtonWrap, this.ccFieldsContainer.querySelector( '[id^=give-card-number-wrap-]' ) );
+		return this.ccFieldsContainer.insertBefore( smartButtonWrap, separator ? separator : cardNumberWarp );
 	}
 
 	/**
 	 * Render smart buttons.
 	 *
 	 * @since 2.9.0
+	 * @return {object} Return Promise
 	 */
 	renderPaymentMethodOption() {
 		this.smartButtonContainer = this.getButtonContainer();
+		const onCancelError = () => DonationForm.addErrors( this.jQueryForm, Give.form.fn.getErrorHTML( [ { message: givePayPalCommerce.defaultDonationCreationError } ] ) );
```

Comment:
@JasonTheAdams What message we can show on the donation form when donor cancels the payment in PayPal popup. I am showing the default message currently.

Reply:
Don't show anything. If they canceled then there wasn't really any error.

Reply:
fixed: https://github.com/impress-org/givewp/compare/d665df914e02...648d8fcb5e29